src/solvers/lowering/byte_operators.cpp

```diff
@@ -17,20 +19,106 @@ Author: Daniel Kroening, kroening@kroening.com
 #include <util/replace_symbol.h>
 #include <util/simplify_expr.h>
 
-#include "flatten_byte_operators.h"
+static exprt unpack_rec(
+  const exprt &src,
+  bool little_endian,
+  const exprt &offset,
+  const exprt &max_bytes,
+  const namespacet &ns,
+  bool unpack_byte_array = false);
+
+/// rewrite an array or vector into its individual bytes
+/// \param src: array/vector to unpack
+/// \param src_size: array/vector size
+/// \param element_width: bit width of array/vector elements
+/// \param little_endian: true, iff assumed endianness is little-endian
+/// \param offset: if not nil, use as lower bound for non-nil bytes to return
+/// \param max_bytes: if not nil, use as upper bound of the number of bytes to
+///                   unpack
+/// \param ns: namespace for type lookups
+/// \param [out] dest_array: target array_exprt to hold unpacked elements
+static void unpack_array_vector(
+  const exprt &src,
+  const exprt &src_size,
+  const mp_integer &element_width,
+  bool little_endian,
+  const exprt &offset,
+  const exprt &max_bytes,
+  const namespacet &ns,
+  array_exprt &dest_array)
+{
+  mp_integer num_elements;
+  if(to_integer(max_bytes, num_elements) && to_integer(src_size, num_elements))
+  {
+    throw "cannot unpack array/vector of non-const size:\n" +
```

Comment:
Again, invariant?

Reply:
It's something that may be fixable in some way, and really a limitation of the implementation rather than a bug. So I'm inclined to leave it in place.